RA-1634 : Update MetaData Packages in the MetaData Module

mozzy11 opened the pull request:
Ticket : https://issues.openmrs.org/browse/RA-1634
Update Metadata Sharing packages in the Reference Metadata module

dkayiwa commented on api/src/main/java/org/openmrs/module/referencemetadata/ReferenceMetadataActivator.java:
```diff
@@ -93,7 +93,7 @@ private void installConcepts() {
         	
         	DataImporter dataImporter = Context.getRegisteredComponent("dataImporter", DataImporter.class);
-            dataImporter.importData("Reference_Application_Concepts-24.xml");
+            dataImporter.importData("Reference_Application_Concepts-26.xml");
```
Any reason why this is 26 instead of 25?

mozzy11 replied:
sorry , it was a mistake

mozzy11 replied:
corrected it

dkayiwa commented:
Are these the only packages that changed?

mozzy11 commented:
yes , they are the only that am required to update as per this 
https://wiki.openmrs.org/display/docs/OpenMRS+Reference+Application+Release+Process

mozzy11 commented:
@dkayiwa  , waiting for your comments

dkayiwa commented:
Which openmrs platform version did you use to generate these changes on the mdsbuilder server?

mozzy11 commented:
@dkayiwa   mdsBuilder runs on platform 2.0.6 , so that what used

dkayiwa commented:
And what is the name of the CIEL file that you used?

mozzy11 commented:
openmrs_concepts_2.0_20190701.sql

dkayiwa commented:
And which OpenMRS platform version will the reference application that you are releasing be running on?

mozzy11 commented:
platform 2.2. Does that mean we have to upgrade the platform version for the mdsBuilderServer ?

mozzy11 commented:
ok . can you do that for me then? i dont have permisons to do that

dkayiwa commented:
Do you have ssh access to that server?

mozzy11 commented:
yes i do .

dkayiwa commented:
So what kind of permissions are you looking for?

mozzy11 commented:
oh i ddnt know i could also upgrade the platform with ssh access. On which server is it hosted , i mean if i can do it via UI . ,it may be more hectic for me to use commands to upgrade the platform.

mozzy11 commented:
does it have the same directories as our local instances we run with tomcat ??

mozzy11 commented:
the server seems to be already running on pltform 2.2

here i the distro properties file
```
 version=1.0.0
distro.platform=2.2.0
omod.metadatasharing=1.3.0
omod.metadatamapping=1.3.3
omod.dataexchange=1.3.3
omod.legacyui=1.3.3

```

mozzy11 commented:
@dkayiwa  i updated the Metadata Packages ,using  **openmrs_concepts_2.2_20191021.sql**  for platform 2.2.
I updated some module versions in the POM , because i was getting alot datase of errors , so i solved thatby updating the Core version dependence to 2.2 and some other module versions

dkayiwa commented:
Did you see the travis build failure?

mozzy11 commented:
@dkayiwa  , resolved the error

mozzy11 commented:
@dkayiwa  , any comments here ?

mozzy11 commented:
ping @dkayiwa

mozzy22 commented:
@dkayiwa  , some look into this

dkayiwa commented on api/src/test/resources/BuiltInReportsTestDataset.xml:
```diff
@@ -114,7 +114,7 @@
 
 	<obs obs_id="1" person_id="52" concept_id="1" encounter_id="1" obs_datetime="2015-06-18 22:21:32" location_id="1" value_coded="1" creator="1"
-	     date_created="2017-06-18 23:45:34" voided="0" uuid="8d3a3ea9-c8fd-4b7b-b6a8-6410500ad410"
+	     date_created="2017-06-18 23:45:34" voided="0" uuid="8d3a3ea9-c8fd-4b7b-b6a8-6410500ad410" status="FINAL"
```
Was the status necessary?

mozzy22 replied:
yes , since i updated most of the modules , it is now required, it cant be null

dkayiwa replied:
What error do you get when you do not touch it?

mozzy22 replied:
some thing like "column OBS.STATUS can not be null."

mozzy22 replied:
i have also confirmed in an actual running database , "column OBS.STATUS" cant be NULL

dkayiwa commented:
After running these changes, do you have the `Head circumference` concept in the dictionary?

mozzy22 commented:
oh well i havent run them locally yet, let me run them and i see

mozzy22 commented:
@dkayiwa  , its not there. However it exists in the Dictionary on the mdsBuilder. Is there a criteria that the  Metadata Sharing module follows to select out concepts to be included in the  package for Reference Application Concepts ??

dkayiwa commented:
Before you export new package versions on mdsbuilder, you need to select and include.

mozzy22 commented:
@dkayiwa ,  i have included the Head Circumfrenece Cocept

dkayiwa commented:
@mozzy11 i mean adding it during the new package version process of the user interface.

mozzy22 commented:
yes @dkayiwa  , thats where i included it in when i was creating the new package, before i exported the package

dkayiwa commented:
Have you now compiled, run the module, and confirmed it now exists?

mozzy22 commented:
in other words , i deleted the previous package i had created from the mdsBuider server, and created a new one which includes the Head circumference concept

dkayiwa commented:
Is that the only new concept that the new refapp requires?

mozzy22 commented:
well , i would think so. Am not aware of any new feature that would require any new concept as far as the Features Road Map we have worked on is concerned

dkayiwa commented:
Have you now compiled, run the module, and confirmed it now exists?

mozzy22 commented:
its what am doing now ..

mozzy22 commented:
yes @dkayiwa , the Head Circumfrence Concept exist , ive just ran the latest snaphot version of the reff app , and deployed the Module ive compliled

dkayiwa commented:
Is that a fresh installation or an existing one?

mozzy22 commented:
a fresh installation

dkayiwa commented:
Can you also test an existing installation?

mozzy22 commented:
like i deloy the module in an existing installation without first deleting the database ?

dkayiwa commented:
Any instance of the reference application that was running without your changes.

mozzy22 commented:
it doesnt update concepts in an existing installation

dkayiwa commented:
So can you fix that?

mozzy11 commented:
@dkayiwa  i fixed that , i increased the  **METADATA_VERSION**  to 17;

mozzy11 commented:
it now works on an existing installtion

dkayiwa commented:
What is the difference between this pull request and the previous one? https://github.com/openmrs/openmrs-module-referencemetadata/pull/46

mozzy11 commented:
@dkayiwa  , with this PR ,
*  i built and updated the packages in this PR using mdsBuilder server , not manual editing like the other side. That saves alot of human error especially when it comes to mappings and UUIDs , no wonder we got CI errors with the other Pr.
I shud actually close it.

* I built this packages with the latest Updated Dictionary (not sure where some updates are also made in the already existing Concepts) , but its the latest Dictionary anyway
